portfolio/src/main/webapp/script.js

```diff
@@ -21,21 +21,29 @@ function addRandomFact() {
   const randomFact = facts[Math.floor(Math.random() * facts.length)];
 
   // Remove any visible fact
-  document.querySelectorAll('.fact.visible').forEach(element => element.classList.remove('visible'));
+  document.querySelectorAll('.fact.visible')
+      .forEach(element => element.classList.remove('visible'));
 
   // Display the random fact chosen
   randomFact.classList.add('visible');
 }
 
+// Retrieves comments from /data and places them in the DOM
 async function addComments() {
   const response = await fetch('/data');
-  const comment = await response.text();
-  document.getElementById('comments-container').innerHTML = comment;
+  const comments = await response.json();
+  for(var key of Object.keys(comments)){
```

Comment:
Should this first empty out the contents of `comments-container`? I imagine that loading comments multiple times will show multiple copies of them, and this may also be preventing deletion from being reflected on the page.